File: tmp/e2e_frontend_smoke.js
const { chromium } = require('playwright');

async function checkPage(browser, url) {
  const page = await browser.newPage();
  const consoleMessages = [];
  const pageErrors = [];
  page.on('console', (msg) => {
    consoleMessages.push({ type: msg.type(), text: msg.text() });
  });
  page.on('pageerror', (err) => {
    pageErrors.push(String(err));
  });

  let status = null;
  try {
    const response = await page.goto(url, { waitUntil: 'networkidle', timeout: 30000 });
    status = response ? response.status() : null;
    await page.waitForTimeout(1500);
    const title = await page.title();
    return {
      url,
      status,
      title,
      pageErrors,
      severeConsole: consoleMessages.filter((item) => item.type === 'error'),
    };
  } finally {
    await page.close();
  }
}

(async () => {
  const browser = await chromium.launch({ headless: true });
  try {
    const results = [];
    results.push(await checkPage(browser, 'http://localhost:3002/'));
    results.push(await checkPage(browser, 'http://localhost:3002/chat/new'));
    console.log(JSON.stringify(results, null, 2));
  } finally {
    await browser.close();
  }
})().catch((error) => {
  console.error(error);
  process.exit(1);
});

File: tmp/e2e_frontend_smoke.py
from playwright.sync_api import sync_playwright
import json


def check_page(browser, url: str):
    page = browser.new_page()
    console_messages = []
    page_errors = []
    page.on("console", lambda msg: console_messages.append({"type": msg.type, "text": msg.text}))
    page.on("pageerror", lambda err: page_errors.append(str(err)))

    response = page.goto(url, wait_until="networkidle", timeout=30000)
    page.wait_for_timeout(1500)
    title = page.title()
    result = {
        "url": url,
        "status": response.status if response else None,
        "title": title,
        "pageErrors": page_errors,
        "severeConsole": [item for item in console_messages if item["type"] == "error"],
    }
    page.close()
    return result


with sync_playwright() as p:
    browser = p.chromium.launch(headless=True)
    try:
        results = [
            check_page(browser, "http://localhost:3002/"),
            check_page(browser, "http://localhost:3002/chat/new"),
        ]
        print(json.dumps(results, ensure_ascii=False, indent=2))
    finally:
        browser.close()
